MAGEDOC-3469: Remove EOL docs

jeff-matthews opened the pull request:
## Purpose of this pull request

This pull request (PR) removes all `guides/v2.0` content from the devdocs site since 2.0 has reached end-of-life (EOL) and no longer receives security patches, quality fixes, or documentation updates.

This PR also replaces `v2.0` in redirect metadata with `v2.3` so that S3 can route URLs to the right file since we're configuring S3 to replace `v2.0` in ALL inbound URLS with `v2.3`. The file-based redirects in the file metadata need to be updated for this solution to work.

**This PR cannot be merged until we receive confirmation that the customer email has been sent.**

## Affected docs

- https://devdocs.magento.com/guides/v2.0/

## Testing

See [staging build](https://devdocs.magedevteam.com/407/) to test the following scenarios:

- All 2.0 URLs
- 2.0 redirects specified in file metadata

### All 2.0 URLs

1. Navigate to https://devdocs.magedevteam.com/407/guides/v2.0/install-gde/bk-install-guide.html or any other 2.0 URL.
2. Confirm that the URL redirects to the same 2.3 file (unless it has a file-based redirect).

### 2.0 redirects from metadata

1. Open [replace-with-2.3.xlsx](https://github.com/magento/devdocs/files/3162765/replace-with-2.3.xlsx).
2. Copy/paste a 2.0 URL in a browser.
3. Confirm that it redirects to the corresponding 2.3 file.

dshevtsov commented:
@jeff-matthews, XLSX is a format that requires a proprietary software to open it. Could you please replace it with CSV or ODS if possible?

jeff-matthews commented:
GitHub doesn't accept CSV or ODS formats.

dshevtsov commented:
Good to know

dshevtsov commented on guides/v2.1/architecture/archi_perspectives/components/modules/mod_and_areas.md:
```diff
@@ -3,7 +3,7 @@ group: architecture-guide
 menu_title: Modules and areas
 redirect_from:
-  - /guides/v2.0/architecture/modules/mod_and_areas.html
+  - /guides/v2.3/architecture/modules/mod_and_areas.html
```
This (and similar in files below) redirects a user form 2.3 link to the 2.1 topic. Is it expected behavior?

dshevtsov replied:
I guess this file is a source for a symlink in 2.3

jeff-matthews replied:
yes, unfortunately, it's a side effect of symlinks. in symlinked files, whatever entry comes first in the metadata wins.

jeff-matthews commented:
I just spoke to @dshevtsov about an idea. What do you all think about doing the following for redirects instead?

1. Keep the 2.0 -> 2.3 replacement on S3
2. Remove **all** 2.0 file-based redirects
3. If a 2.0 file doesn't exist in 2.3 and causes a 404, redirect to the 2.3 devdocs home page

This is a lot easier and has the added benefit of removing old redirects to 2.0 content that likely receives very little _direct_ traffic anyway.

hguthrie commented:
I say we send them to the website to buy a new Cloud account.

jeff-matthews commented:
@dobooth, good catch. Not all are related to magento. At least one is related to Elasticsearch. The magento-specific references don't use a variable like they should. I can do a search and replace, but it would require verifying that the file still exists in the latest codebase.

jeff-matthews commented:
I reverted the commit that globally replaced 2.0 redirects in file metadata (4361cac).

jeff-matthews commented:
Ok, I removed all 2.0 redirects from the file metadata and updated the S3 config on staging.

If a 2.3 file exists for an incoming 2.0 URL, S3 will redirect to that page. For example:

> https://devdocs.magedevteam.com/415/guides/v2.0/install-gde/bk-install-guide.html

If a 2.3 file does not exist for an incoming 2.0 URL, S3 will redirect to the home page of the latest version of docs (currently 2.3). For example:

> https://devdocs.magedevteam.com/415/guides/v2.0/this-file-does-not-exist.html

jeff-matthews commented:
Thanks for the review and improvements @dobooth.

Does anyone else want to offer an opinion on this before I put it in motion?

jeff-matthews commented:
Good point @dshevtsov. I'll make the update.

jeff-matthews commented:
running tests

jeff-matthews commented:
running tests

jeff-matthews commented:
Tests failed

```
+ rake test:cicd
[35mChecking links with htmlproofer...[0m
Running ["LinkChecker::DoubleSlashCheck", "ImageCheck", "ScriptCheck", "LinkCheck"] on ["_site"] on *.html... 


Ran on 4070 files!


- internally linking to /guides/v2.0/cloud/release-notes/CloudReleaseNotes2.1.3.html#cloud-es-config-mg, which does not exist
  *  _site/guides/v2.1/cloud/live/sens-data-initial.html (line 467)
     <a href="/guides/v2.0/cloud/release-notes/CloudReleaseNotes2.1.3.html#cloud-es-config-mg">Find Admin login information</a>
- internally linking to /guides/v2.0/ui-components/ui_components_js.html#comp_link, which does not exist
  *  _site/guides/v2.1/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
  *  _site/guides/v2.2/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
  *  _site/guides/v2.3/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
- internally linking to app/design/adminhtml/Magento/backend/web/fonts/admin-icons, which does not exist
  *  _site/guides/v2.1/design-styleguide/iconography/iconography.html (line 274)
     <a href="app/design/adminhtml/Magento/backend/web/fonts/admin-icons">in GitHub</a>
  *  _site/guides/v2.2/design-styleguide/iconography/iconography.html (line 274)
     <a href="app/design/adminhtml/Magento/backend/web/fonts/admin-icons">in GitHub</a>
  *  _site/guides/v2.3/design-styleguide/iconography/iconography.html (line 274)
     <a href="app/design/adminhtml/Magento/backend/web/fonts/admin-icons">in GitHub</a>
- internally linking to lib/internal/Magento/Framework/Code/Minifier, which does not exist
  *  _site/guides/v2.1/extension-dev-guide/adapters.html (line 230)
     <a href="lib/internal/Magento/Framework/Code/Minifier" target="_blank">minifier</a>
  *  _site/guides/v2.2/extension-dev-guide/adapters.html (line 230)
     <a href="lib/internal/Magento/Framework/Code/Minifier" target="_blank">minifier</a>
  *  _site/guides/v2.3/extension-dev-guide/adapters.html (line 230)
     <a href="lib/internal/Magento/Framework/Code/Minifier" target="_blank">minifier</a>
- internally linking to lib/internal/Magento/Framework/Image, which does not exist
  *  _site/guides/v2.1/extension-dev-guide/adapters.html (line 238)
     <a href="lib/internal/Magento/Framework/Image" target="_blank"></a>
  *  _site/guides/v2.2/extension-dev-guide/adapters.html (line 238)
     <a href="lib/internal/Magento/Framework/Image" target="_blank"></a>
  *  _site/guides/v2.3/extension-dev-guide/adapters.html (line 238)
     <a href="lib/internal/Magento/Framework/Image" target="_blank"></a>
- internally linking to setup/performance-toolkit, which does not exist
  *  _site/guides/v2.1/cloud/live/stage-prod-test.html (line 410)
     <a href="setup/performance-toolkit">Magento Performance Toolkit</a>
  *  _site/guides/v2.2/cloud/live/stage-prod-test.html (line 410)
     <a href="setup/performance-toolkit">Magento Performance Toolkit</a>
  *  _site/guides/v2.3/cloud/live/stage-prod-test.html (line 410)
     <a href="setup/performance-toolkit">Magento Performance Toolkit</a>
- trying to find hash of /guides/v2.0/cloud/release-notes/CloudReleaseNotes2.1.3.html#cloud-es-config-mg, but /apps/slave/workspace/devdocs-test/_site/guides/v2.0/cloud/release-notes/CloudReleaseNotes2.1.3.html does not exist
  *  _site/guides/v2.1/cloud/live/sens-data-initial.html (line 467)
     <a href="/guides/v2.0/cloud/release-notes/CloudReleaseNotes2.1.3.html#cloud-es-config-mg">Find Admin login information</a>
- trying to find hash of /guides/v2.0/ui-components/ui_components_js.html#comp_link, but /apps/slave/workspace/devdocs-test/_site/guides/v2.0/ui-components/ui_components_js.html does not exist
  *  _site/guides/v2.1/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
  *  _site/guides/v2.2/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
  *  _site/guides/v2.3/ui_comp_guide/concepts/ui_comp_uielement_concept.html (line 214)
     <a href="/guides/v2.0/ui-components/ui_components_js.html#comp_link">, ,  and  properties</a>
rake aborted!
HTML-Proofer found 20 failures!```

jeff-matthews commented:
running tests
